prowler/providers/aws/lib/s3/s3.py

```diff
@@ -147,3 +150,38 @@ def send_to_bucket(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}] -- {error}"
             )
         return uploaded_objects
+
+    @staticmethod
+    def test_connection(
+        session, bucket_name: str, raise_on_exception: bool = True
+    ) -> Connection:
+        """
+        Test the connection to the S3 bucket.
+
+        Parameters:
+        - session: An instance of the `Session` class representing the AWS session.
+        - bucket_name: A string representing the name of the S3 bucket.
+        - raise_on_exception: A boolean indicating whether to raise an exception if the connection test fails.
+
+        Returns:
+        - A boolean indicating whether the connection to the S3 bucket was successful.
+
+        Raises:
+        - Exception: An exception indicating that the connection test failed.
+        """
+        try:
+            # Set a Temp file to upload
+            with tempfile.TemporaryFile() as temp_file:
+                temp_file.write(b"Test Prowler Connection")
+                temp_file.seek(0)
+                session.upload_fileobj(
+                    temp_file, bucket_name, "test-prowler-connection.txt"
+                )
+
+            # Try to delete the file
+            session.delete_object(Bucket=bucket_name, Key="test-prowler-connection.txt")
+            return True
+        except Exception as error:
```

Comment:
We need to catch exceptions here coming from S3 and create our own.

Reply:
Done